### ISSUE TITLE
Comments for Installing PostgreSQL without Root

### ISSUE THREAD

**Issue by phinjensen** [EndPointCorp/end-point-blog#819]
Comments for https://www.endpointdev.com/blog/2013/06/installing-postgresql-without-root/
By Szymon Lipiński

To enter a comment:
1. Log in to GitHub
2. Leave a comment on this issue.

**Comment by Nopony** [comment]
Worked like a charm, saved me a lot of time. Thanks.

**Comment by kanihal** [comment]
Thanks for the tutorial. It was Very helpful.

**Comment by heckur** [comment]
I followed the steps and everything worked fine except for the last part, my system user was not created in the db:

`bin/psql -U my_linux_user postgres
psql: FATAL:  role "my_linux_user" does not exist`

and it won't let me create it:

`./bin/createuser --interactive
Enter name of role to add: my_linux_user
Shall the new role be a superuser? (y/n) y
createuser: could not connect to database postgres: FATAL:  role "my_linux_user" does not exist`

since I don't have root privileges, and the user postgres does not exist in the system, I don't know in what other way I can create my role, any ideas?

**Comment by machack666** [comment]
Hi, does it work using when adding the `-U postgres` flag?

**Comment by heckur** [comment]
> Hi, does it work using when adding the `-U postgres` flag?

No, the postgres user was not created either so I get the same error:

    bin/psql -U postgres postgres
    psql: FATAL:  role "postgres" does not exist

**Comment by heckur** [comment]
It was my fault, incorrect postgres binaries (initdb, createuser...) were in my LD_LIBRARY_PATH, so they were trying to connect me to a faulty installation I guess, once I removed them and added the correct ones, I was able to connect to a db that already had my system user created.

**Comment by madhu1agrawal** [comment]
I am getting the error "Is the server running locally and accepting	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"? How can I resolve this? This happens when run postgres/bin/psql -U <username> postgres.

**Comment by theaaron123** [comment]
> I am getting the error "Is the server running locally and accepting connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"? How can I resolve this? This happens when run postgres/bin/psql -U postgres.

For future reference. 

Try `postgres/bin/psql -U postgres -h /tmp`  so it connects through the correct domain socket. I had to do this on RHEL7.
